scalyr_agent/log_processing.py

```diff
@@ -2417,12 +2417,10 @@ def perform_processing(self, add_events_request, current_time=None):
                     if self._log_stream:
                         import scalyr_ingestion_client.log_line as ingestion_client_line  # pylint: disable=import-error
 
-                        if not new_events_buffer:
-                            sequence_start_number = sequence_number
                         sequence_end_number = sequence_number
                         new_event_timestamp = line_object.timestamp
                         if not new_event_timestamp:
-                            new_event_timestamp = int(current_time)
+                            new_event_timestamp = int(current_time * 1000000000)
```

Comment:
You could also use new ``scalyr_ingestion_client.time_utils.get_nanoseconds_from_epoch`` method.

Reply:
That method would get us a slightly different timestamp for each event as we processed through them, if we have no parsed timestamp to work with the expected behavior is that we would have the same timestamp for every event right? Using that method elsewhere in the function to get a nanosecond current time would also require another conditional import so I think using the existing `current_time` is our best bet, though I did move this calculation to the beginning fo the method.